### PR TITLE
add LPASS_PASSFD environment variable

### PR DESCRIPTION
**Pull request by xdgc**
If LPASS_PASSFD evaluates to a nonzero integer, password
will be read from that file descriptor.

Signed-off-by: David Champion <dgc@bikeshed.us>

### CODE REVIEW

File: password.c [hunk 1]
```diff
@@ -231,6 +231,8 @@ char *password_prompt(const char *prompt, const char *error, const char *descfmt
 	char *ret;
 	va_list params;
 	int devnull;
+    char *passfd_s;
+    int passfd;
```
Comment (bcopeland) [inline password.c]: whitespace issues here

File: password.c [hunk 2]
```diff
@@ -240,6 +242,24 @@ char *password_prompt(const char *prompt, const char *error, const char *descfmt
 		return askpass;
 	}
 
+    passfd_s = getenv("LPASS_PASSFD");
+    if (passfd_s) {
+        passfd = atoi(passfd_s);
+        if (passfd > 0) {
+            char *password = NULL;
+            int pwlen = 0, off = 0, incr = 64, more = 64;
```
Comment (bcopeland) [inline password.c]: Can just use getline() here, like the other cases.

**Comment by bcopeland** [comment]
I appreciate that custom askpass or pinentry is complicated, but this won't work for anyone with a second factor.  Also, have you considered doing just:
```
function login_to_lpass {
    export LPASS_DISABLE_PINENTRY=1
    echo -n "mypassword" | ./lpass login myaccount@example.com 2>/dev/null
}
```
Besides that, have you considered using the ```--plaintext-key``` option?  This option allows you to login once to lpass, and then just read the key from the file system going forward.

**Comment by bcopeland** [comment]
I should say, before you spend time rewriting things to address my inline comments, that I'm a bit reticent to add a fourth way to accept passwords -- it would be great if we could find a way to do what you want with the existing machinery.

**Comment by xdgc** [comment]
OK, thanks for the review. I realize it won't work for MFA, but the goal (for me) is to make it work with unattended automations, and MFA is unlikely in that scenario anyway — though possible, I guess. I'm typically running without the agent because in an automation I don't want anything cached locally, I want it to sync immediately. So "lpass login" doesn't happen.

It's entirely possible that I'm missing some information on how I can do this but the documentation only addresses basic use cases, and I haven't been able to read all the code yet.

**Pull request closed without merging**